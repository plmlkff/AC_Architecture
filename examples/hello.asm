WORD: word 'h'
    word 'e'
    word 'l'
    word 'l'
    word 'o'
    word ' '
    word 'w'
    word 'o'
    word 'r'
    word 'l'
    word 'd'
POINTER: word WORD
LENGTH: word 11
START:
    ld [POINTER]
    wr 1022
    ld POINTER
    add #1
    cmp LENGTH
    je #.end
    wr POINTER
    jmp #START
.end:
    hlt
