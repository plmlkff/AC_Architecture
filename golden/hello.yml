in_source: |
  WORD: word 'h'
      word 'e'
      word 'l'
      word 'l'
      word 'o'
      word ' '
      word 'w'
      word 'o'
      word 'r'
      word 'l'
      word 'd'
  POINTER: word WORD
  LENGTH: word 11
  START:
      ld [POINTER]
      wr 1022
      ld POINTER
      add #1
      cmp LENGTH
      je #end
      wr POINTER
      jmp #START
  end:
      hlt
in_stdin: |-
  5Trash
out_code: |-
  [104, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100, 0, 11, {"name": "ld", "op_code": "LD", "addressing_type": "INDIRECT_STRAIGHT", "is_start": true, "address": 11}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 1022}, {"name": "ld", "op_code": "LD", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 11}, {"name": "add", "op_code": "ADD", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 1}, {"name": "cmp", "op_code": "CMP", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 12}, {"name": "je", "op_code": "JE", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 21}, {"name": "wr", "op_code": "WR", "addressing_type": "ABSOLUTE_STRAIGHT", "is_start": false, "address": 11}, {"name": "jmp", "op_code": "JMP", "addressing_type": "DIRECT_LOAD", "is_start": false, "address": 13}, {"name": "hlt", "op_code": "HLT", "addressing_type": "NO_ADDRESS", "is_start": false, "address": 0}, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
out_stdout: |
  code.o
  ============================================================
  hello world
  Ticks count: 2804
out_log: |
  DEBUG   Machine:tick_log      Tick #2766:  mIP: 52; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 17; AR: 12; N: False; Z: False
  DEBUG   Machine:tick_log      Tick #2767:  mIP: 53; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 17; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2768:  mIP: 69; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 17; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2769:  mIP: 70; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2770:  mIP: 0; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 12; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2771:  mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2772: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2773: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2774: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2775: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2776: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2777: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2778: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2779: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2780: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2781: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 11; SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2782:  mIP: 2; AC: 11; BR: 0; DR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); SP: -3; CR: Command(name='cmp', op_code=<OpCode.CMP: 52>, addressing_type=<AddressingType.ABSOLUTE_STRAIGHT: 0>, is_start=False, address=12); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2783:  mIP: 3; AC: 11; BR: 0; DR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2784:  mIP: 16; AC: 11; BR: 0; DR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2785:  mIP: 17; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2786:  mIP: 56; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2787:  mIP: 57; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 18; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2788:  mIP: 58; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2789:  mIP: 0; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 18; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2790:  mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2791: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2792: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2793: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2794: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2795: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2796: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2797: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2798: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2799: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2800: MEMORY ACCESS TICK mIP: 1; AC: 11; BR: 0; DR: 21; SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2801:  mIP: 2; AC: 11; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='je', op_code=<OpCode.JE: 56>, addressing_type=<AddressingType.DIRECT_LOAD: 4>, is_start=False, address=21); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2802:  mIP: 3; AC: 11; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2803:  mIP: 19; AC: 11; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 21; AR: 21; N: False; Z: True
  DEBUG   Machine:tick_log      Tick #2804:  mIP: 68; AC: 11; BR: 0; DR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); SP: -3; CR: Command(name='hlt', op_code=<OpCode.HLT: 68>, addressing_type=<AddressingType.NO_ADDRESS: 5>, is_start=False, address=0); IP: 21; AR: 21; N: False; Z: True
